rest/src/main/java/discord4j/rest/util/ApplicationCommandType.java

```diff
@@ -0,0 +1,69 @@
+/*
+ *  This file is part of Discord4J.
+ *
+ * Discord4J is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Discord4J is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Discord4J.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package discord4j.rest.util;
+
+/**
+ * @see <a href="https://discord.com/developers/docs/interactions/application-commands#application-command-object-application-command-types">
+ * Application Command Type</a>
+ */
+public enum ApplicationCommandType {
```

Comment:
Should I also move `ApplicationCommandOptionType` to be nested into the option class? I put this in rest.util since the option type enum was already there.